### PR TITLE
fix(profiling): better autoinject detection

### PR DESCRIPTION
**Pull request by sanchda**
[PROF-10280]

Previously, Profiling was injecting its own configuration during lib-injection handling.  This was actually a little brittle.

Instead, we move to a model where the profiler detects (in its normal configuration) whether the library was injected.  Hopefully this is more durable and correct.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

[PROF-10280]: https://datadoghq.atlassian.net/browse/PROF-10280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sanchda** [comment]
* **#9904** <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9904?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sanchda and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ddtrace/settings/profiling.py [hunk 1]
```diff
@@ -315,6 +362,12 @@ class ProfilingConfigExport(En):
 
 config = ProfilingConfig()
 
+# If during processing we discover that the configuration was injected, we need to do a few things
+# - Mark it as such
+# - Force libdd to be enabled, disabling the profiler otherwise the service might crash
+#   (this is done in the _is_libdd_required function)
+config.injected = _check_for_injected()
```
Comment (Kyle-Verhoog) [inline ddtrace/settings/profiling.py:369]:
```suggestion
config._injected = _check_for_injected()
```
no need to leak to the public API?

**Review by Kyle-Verhoog — COMMENTED** [review]
just one thought, otherwise good to me!

File: ddtrace/settings/profiling.py (hunk 1 above)
Comment (sanchda) [reply ddtrace/settings/profiling.py]: Good idea, fixed this and the references

**Review by Kyle-Verhoog — APPROVED** [review]

File: tests/lib-injection/dd-lib-python-init-test-protobuf-old/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+protobuf==3.17.0
```
[the remark this answers is not in the document]
Comment (sanchda) [reply tests/lib-injection/dd-lib-python-init-test-protobuf-old/requirements.txt:1]: This is intentional.

**Review by taegyunkim — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**